client/components/theme-collection/index.tsx

```diff
@@ -0,0 +1,133 @@
+import { Button } from '@wordpress/components';
+import { Icon, chevronLeft, chevronRight } from '@wordpress/icons';
+import { ReactElement, useEffect, useRef } from 'react';
+import './style.scss';
+import { Swiper as SwiperType } from 'swiper/types';
+import { ThemeBlock } from 'calypso/components/themes-list';
+
+interface ThemeCollectionProps {
+	heading: string;
+	subheading: ReactElement;
+	themes: Array< { id: string } >;
+	collectionSlug: string;
+	getScreenshotUrl: ( themeId: string ) => string;
+	siteId: string | null;
+	getButtonOptions: () => void;
+	getActionLabel: () => string;
+	isActive: () => boolean;
+	getPrice: () => string;
+	isInstalling: () => boolean;
+}
+
+export default function ThemeCollection( {
+	collectionSlug,
+	heading,
+	subheading,
+	themes,
+	getScreenshotUrl,
+	siteId,
+	getButtonOptions,
+	getActionLabel,
+	isActive,
+	getPrice,
+	isInstalling,
+}: ThemeCollectionProps ): ReactElement {
```

Comment:
Nitpick: let's try to keep props ordered alphabetically (in all component functions and calls).